### ISSUE TITLE
Explore Slack integrations for deployment

### ISSUE THREAD

**Issue by alanmoo** [MozillaFoundation/mofo-devops#543]
It would be great if we could deploy/promote apps within Slack, so it's immediately visible to everyone on the team. There's probably some other neat things we could do, so let's explore.

**Comment by patjouk** [comment]
I took a look at Heroku docs about their [ChatOps](https://devcenter.heroku.com/articles/chatops) and it seems pretty cool.
I can give it a try: do you already have an app in mind that I could use for testing? The only limitation is that it must be a pipeline.

**Comment by alanmoo** [comment]
Heh, important limitation right now. @mmmavis Does Mozfest use pipelines? I feel like that's a safe one to test this with because it doesn't have database migrations to worry about

**Comment by mmmavis** [comment]
Bad news! @alanmoo MozFest site does have pipeline set up but I've never used "promote to prod" button for prod push (I always manually push). I just tried that button and it didn't seem to update production... 😓

**Comment by mmmavis** [comment]
Sorry, false alarm in my previous comment. I've verified that pipeline does work for MozFest deployment.

**Comment by patjouk** [comment]
It was pretty easy to setup and should now be working.

**What you need to do:**

<img width="610" alt="screen shot 2017-10-23 at 2 59 50 pm" src="https://user-images.githubusercontent.com/481052/31890323-ff17b5b2-b802-11e7-8821-e0d02b425ee6.png">

You have to login on Heroku using `/h login` on Slack. It will automatically send a message, asking you to `sign in to Github`. Click on the link in the message and follow the instructions.
If you get the `You're all set` message, you're good to go. If you have an error, [verify if you have the good permissions](https://devcenter.heroku.com/articles/chatops#permissions).

**Relevant commands on Slack:**
When you want to interact with the ChatOps bot, use `\h [your command]` pattern.
- `/h help`: list all the available commands.
- `/h pipelines`: list all the available pipelines with a link to their dashboard.
- `/h promote PIPELINE_NAME`. Ex: `/h promote mozillafestival-org`: will promote staging to prod.
- `/h deploy PIPELINE_NAME to STAGE_NAME`. Ex: `/h deploy mozillafestival-org to staging`: will deploy master to staging.
- `/h releases PIPELINE_NAME`. Ex: `/h releases mozillafestival-org`: list the latest releases from prod. Add `in staging` to get the latest from staging.

Of course, that's just a few things that the bot can do: you can find the [full docs and examples here](https://devcenter.heroku.com/articles/chatops).

**Comment by alanmoo** [comment]
Given that we need a security review before granting ChatOps write access to our repos, I wonder if there's another approach that doesn't, like Hubot or something like that. (Though I do like the feel of ChatOps, especially since it's built by Heroku for exactly the purposes I described)

**Comment by cadecairos** [comment]
_see the [Bugzilla bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1411002) for latest updates_

**Comment by cadecairos** [comment]
@patjouk and I think that the right way to go, now that ChatOps isn't happening, is to write a custom hubot that can deploy specific apps. This isn't a priority, and thus has been labelled "rainy-day-project". Feel free to switch onto this project if you need to get away from whatever task you're stuck on.

* Set up repo
* scaffolding for the hubot app
* write the integrations

**Comment by patjouk** [comment]
Update on that: we gonna wait for ChatOps with new GitHub permissions to be out of alpha.

**Comment by cadecairos** [comment]
We've made some good strides using Slack integrations. We're waiting on beta access for an updated version of ChatOps. Until then there's nothing more to do here, so lets create a new issue when we can proceed.
